admin-auth/src/test/kotlin/com/epam/drill/admin/auth/TestApplication.kt

```diff
@@ -72,4 +78,42 @@ fun testModule(
     }
 }
 
+fun TestApplicationEngine.withStatusPages() {
+    callInterceptor = {
+        try {
+            call.application.execute(call)
+        } catch (_: Throwable) {
+            //allow to handle status pages
+        }
+    }
+}
+
+fun TestApplicationRequest.addBasicAuth(username: String, password: String) {
+    val encodedCredentials = String(Base64.getEncoder().encode("$username:$password".toByteArray()))
+    addHeader(HttpHeaders.Authorization, "Basic $encodedCredentials")
+}
+
+fun TestApplicationRequest.addJwtToken(
+    username: String,
+    secret: String,
+    expiresAt: Date = Date(System.currentTimeMillis() + 10_000),
+    role: String = Role.UNDEFINED.name,
+    issuer: String? = "test issuer",
+    audience: String? = null
+) {
+    val token = JWT.create()
+        .withSubject(username)
+        .withIssuer(issuer)
+        .withAudience(audience)
+        .withClaim("role", role)
+        .withExpiresAt(expiresAt)
+        .sign(Algorithm.HMAC512(secret))
+    addHeader(HttpHeaders.Authorization, "Bearer $token")
+}
+
+fun <T> TestApplicationCall.responseData(serializer: KSerializer<T>): T {
+    val value = assertNotNull(response.content)
```

Comment:
Do not place assertions inside test framework functions* - that makes it harder to comprehend tests + tests may break in unpredictive manner
*Unless these functions are clearly named "assert/Something/" - e.g. `"assertResponseNotNull"` in this case